File: src/micapolos/zexy/Mouse.java
package micapolos.zexy;

import static micapolos.tata8.Game.*;
import static micapolos.zexy.Number.*;
import static micapolos.zexy.Position.*;

public final class Mouse implements Showable {
  public final Position position =
    position(
      number(() -> mouse.position.x),
      number(() -> mouse.position.y));

  public final Boolean isPressed = Boolean.bool(mouse.button::isPressed);
  public final Event press = Event.event(mouse.button::didPress);

  @Override
  public String toString() {
    return String.format("mouse(%s, %s)", position, isPressed);
  }

  static void main() {
    new Mouse().show();
  }
}

File: src/micapolos/zexy/Mouse.kt
package micapolos.zexy

import micapolos.zexy.model.Integer as ModelInteger

object Mouse {
  val position = position(
    Integer(ModelInteger.Apply0(ModelInteger.Op0.MOUSE_X)),
    Integer(ModelInteger.Apply0(ModelInteger.Op0.MOUSE_Y))
  )

  val isPressed = Bool(ModelInteger.Apply0(ModelInteger.Op0.MOUSE_DOWN))
}

val mouse = Mouse